### PR TITLE
add pagination to contributor modal

### PR DESCRIPTION
**Pull request by Ripplz**
Loom -

https://www.loom.com/share/31e94a84a9ea4bbbb2db8e283b4bd104

Wonder Task - https://app.wonderverse.xyz/organization/wonderverse/boards?task=60920008746730267

Task Description:

- right now if you click on contributors on an org/pod page, it brings up this modal. It would be great to add pagination and search to the contributor list, for now, we can just do virtual pagination. don’t really have to do backend pagination to make searching easier

What I Implemented:

- optimized styling of contributors/pods list modal and children items/components to match Figma design.
- added search UI and functionality for contributors and pods lists; searching functionality works with both contributor name & bio, and pod name and description.
- virtual pagination for contributors and pods lists, with added optimizations to make the scrollbar disappear when list is being scrolled.

### CODE REVIEW

**Review by andywong418 — COMMENTED** [review]
@Ripplz could you make a Loom and add a description for what this does (linking Wonder task as well)?

File: wondrous-app/components/profile/modals/index.tsx [hunk 1]
```diff
@@ -1,17 +1,33 @@
...
-import { useInView } from 'react-intersection-observer';
 import { GET_ORG_PODS, GET_ORG_USERS } from 'graphql/queries/org';
 import { GET_POD_USERS } from 'graphql/queries/pod';
 import { useLazyQuery } from '@apollo/client';
-import { TaskModalBaseCard, TaskSubmissionHeaderCreatorText } from '../../Common/Task/styles';
-import { TabContainer, Tab, TabContainerText, PodExplainerText } from './styles';
-import { DefaultProfilePicture, PodWrapper, Title, UserProfilePicture, UserWrapper } from './styles';
+import Image from 'next/image';
+
+import { TaskModalBaseCard } from '../../Common/Task/styles';
```
Comment (Lamperoyge) [inline wondrous-app/components/profile/modals/index.tsx:9]: can we use absolute imports here? e.g import {TaskModalBaseCard} from 'components/Common/Task/styles'

File: wondrous-app/components/profile/modals/index.tsx [hunk 2]
```diff
@@ -1,17 +1,33 @@
...
+  StyledTabs,
+  TabContainerText,
+  TabText,
+  Title,
+  UserProfilePicture,
+  UserWrapper,
+} from './styles';
 import { useRouter } from 'next/router';
-import { CommentText, CommentTopFlexDiv } from '../../Comment/styles';
+import { CommentTopFlexDiv } from '../../Comment/styles';
```
Comment (Lamperoyge) [inline wondrous-app/components/profile/modals/index.tsx:30]: same comment with absolute imports

File: wondrous-app/components/profile/modals/index.tsx [hunk 3]
```diff
@@ -64,26 +80,32 @@ const UserItem = (props) => {
         }}
       >
         <CommentTopFlexDiv>
-          <TaskSubmissionHeaderCreatorText
+          <NameText
```
Comment (Lamperoyge) [inline wondrous-app/components/profile/modals/index.tsx:83]: can we avoid using inline styles?

File: wondrous-app/components/profile/modals/index.tsx [hunk 4]
```diff
@@ -64,26 +80,32 @@ const UserItem = (props) => {
...
   const [displayUsers, setDisplayUsers] = useState(showUsers);
   const [listLoading, setListLoading] = useState(true);
   const [displayPods, setDisplayPods] = useState(showPods);
   const [userList, setUserList] = useState([]);
   const [podList, setPodList] = useState([]);
+  const [userListSearch, setUserListSearch] = useState([]);
+  const [podListSearch, setPodListSearch] = useState([]);
+  const [dividedUserList, setDividedUserList] = useState([]);
+  const [dividedPodList, setDividedPodList] = useState([]);
+  const [activeTab, setActiveTab] = useState('contributors');
```
Comment (Lamperoyge) [inline wondrous-app/components/profile/modals/index.tsx:108]: can you create a config for these vars? e.g TABS_MAP = {CONTRIBUTORS: 'contributors', POD: 'pod'} etc

File: wondrous-app/components/profile/modals/index.tsx [hunk 5]
```diff
@@ -105,12 +126,70 @@ export const MoreInfoModal = (props) => {
...
+        } else if (item.description) {
+          return item.description.toLowerCase().includes(value.toLowerCase());
+        }
+      });
+      setPodListSearch(newList);
+    } else {
+      setPodListSearch(podList);
+    }
+  };
+  const array_chunks = (array, chunk_size) =>
```
Comment (Lamperoyge) [inline wondrous-app/components/profile/modals/index.tsx:168]: can we switch to camelCase? array_chunks => arrayChunks

Also wondering if this function can be refactored ?

File: wondrous-app/components/profile/modals/styles.tsx [hunk 6]
```diff
@@ -65,7 +88,65 @@ export const UserProfilePicture = (props) => {
 export const DefaultProfilePicture = (props) => {
   return <DefaultUserImage style={ProfilePictureStyles} />;
 };
+export const FlexTitle = styled.div`
```
Comment (Lamperoyge) [inline wondrous-app/components/profile/modals/styles.tsx:91]: same with the naming, can we make it more informative?

**Review by Lamperoyge — COMMENTED** [review]
one more q:

I think we already have icons for cancel and search, can we reuse them? 

You can apply different styles by tagging it from styled-components

e.g`
svg {
path: ''
}
`

atm we have a lot of icons and let's try not to add duplicates

**Comment by Ripplz** [comment]
> @Ripplz could you make a Loom and add a description for what this does (linking Wonder task as well)?

Did that already! Here:

Loom - https://www.loom.com/share/31e94a84a9ea4bbbb2db8e283b4bd104

Wonder Task - https://app.wonderverse.xyz/organization/wonderverse/boards?task=60920008746730267

Description: right now if you click on contributors on an org/pod page, it brings up this modal. It would be great to add pagination and search to the contributor list,  for now, we can just do virtual pagination. don’t really have to do backend pagination to make searching easier.

**Comment by Lamperoyge** [comment]
> > @Ripplz could you make a Loom and add a description for what this does (linking Wonder task as well)?
> 
> Did that already! Here:
> 
> Loom - https://www.loom.com/share/31e94a84a9ea4bbbb2db8e283b4bd104
> 
> Wonder Task - https://app.wonderverse.xyz/organization/wonderverse/boards?task=60920008746730267
> 
> Description: right now if you click on contributors on an org/pod page, it brings up this modal. It would be great to add pagination and search to the contributor list, for now, we can just do virtual pagination. don’t really have to do backend pagination to make searching easier.

Thank you! Could you also add this to the PR description ? like here
Screenshot 2022-07-20 at 10 21 22

**Comment by Ripplz** [comment]
> one more q:
> 
> I think we already have icons for cancel and search, can we reuse them?
> 
> You can apply different styles by tagging it from styled-components
> 
> e.g`svg { path: '' }`
> 
> atm we have a lot of icons and let's try not to add duplicates

done

File: wondrous-app/components/profile/modals/index.tsx (hunk 1 above)
Comment (Ripplz) [reply wondrous-app/components/profile/modals/index.tsx]: done

File: wondrous-app/components/profile/modals/index.tsx (hunk 2 above)
Comment (Ripplz) [reply wondrous-app/components/profile/modals/index.tsx]: done

File: wondrous-app/components/profile/modals/index.tsx (hunk 3 above)
Comment (Ripplz) [reply wondrous-app/components/profile/modals/index.tsx]: done

File: wondrous-app/components/profile/modals/index.tsx (hunk 4 above)
Comment (Ripplz) [reply wondrous-app/components/profile/modals/index.tsx]: done

File: wondrous-app/components/profile/modals/index.tsx (hunk 5 above)
Comment (Ripplz) [reply wondrous-app/components/profile/modals/index.tsx]: done

File: wondrous-app/components/profile/modals/styles.tsx (hunk 6 above)
Comment (Ripplz) [reply wondrous-app/components/profile/modals/styles.tsx]: done

**Comment by andywong418** [comment]
@Ripplz Sorry for the delay - could we hide the scrollbar as we scroll down? Looks good after that change!

**Comment by Ripplz** [comment]
> could we hide the scrollbar as we scroll down?

if I understand this correctly, the scrollbar should be hidden * only * when a scroll is happening, and should still be shown at other times - when there's no scroll happening, when the scrollbar is being manually moved with the cursor etc... 

right?

**Comment by Ripplz** [comment]
> @Ripplz Sorry for the delay - could we hide the scrollbar as we scroll down? Looks good after that change!

done. here's what that looks like - https://www.loom.com/share/7a76e9a48a9545b28dc5484709e1cbb8

**Comment by andywong418** [comment]
You can add this to the styles

`  &::-webkit-scrollbar {
    display: none;
    width: 0;
    height: 0;
  }
  /* Hide scrollbar for IE, Edge and Firefox */
  -ms-overflow-style: none; /* IE and Edge */
  scrollbar-width: none; /* Firefox */`

**Comment by Ripplz** [comment]
> You can add this to the styles
> 
> ` &::-webkit-scrollbar { display: none; width: 0; height: 0; } /* Hide scrollbar for IE, Edge and Firefox */ -ms-overflow-style: none; /* IE and Edge */ scrollbar-width: none; /* Firefox */`

done!

**Pull request merged**